consensus/istanbul/backend/backend.go

```diff
@@ -229,13 +229,28 @@ func (sb *backend) checkInSubList(prevHash common.Hash, valSet istanbul.Validato
 }
 
 // getTargetReceivers returns a map of nodes which need to receive a message
-func (sb *backend) getTargetReceivers(valSet istanbul.ValidatorSet) map[common.Address]bool {
+func (sb *backend) getTargetReceivers(prevHash common.Hash, valSet istanbul.ValidatorSet) map[common.Address]bool {
 	targets := make(map[common.Address]bool)
 
-	for _, val := range valSet.List() {
-		if val.Address() != sb.Address() {
-			targets[val.Address()] = true
+	cv := sb.currentView.Load().(*istanbul.View)
+	view := &istanbul.View{
+		Round:    big.NewInt(cv.Round.Int64()),
+		Sequence: big.NewInt(cv.Sequence.Int64()),
+	}
+
+	proposer := valSet.GetProposer()
+	for i := 0; i < 2; i++ {
+		vals := valSet.SubListWithProposer(prevHash, proposer.Address(), view)
```

Comment:
As `SubListWithProposer` leaves below log, calling the function here will leave the same log twice. Is it okay? And also, it would be better to change `vals` to `committee` or `subList` to make it consistent with other codes.

Reply:
Also, how about introducing the method `SubListWithoutProposer` needed for this case?

Reply:
> As `SubListWithProposer` leaves below log, calling the function here will leave the same log twice. Is it okay? And also, it would be better to change `vals` to `committee` or `subList` to make it consistent with other codes.
> 
> ```
> logger.Trace("New committee", "valSet.Number", valSet.blockNum, "prevHash", prevHash.Hex(), "proposer", proposer, "committee", committee, "committee size", len(committee), "subSize", valSet.subSize)
> ```

I agree with that the log can mislead people who is reading a log file. I'll discuss this with @laggu
And changing the name of the variable is a good idea. I'll apply it. Thanks!

Reply:
> Also, how about introducing the method `SubListWithoutProposer` needed for this case?
> It would provide more semantic understanding of the code.

There is a function `SubList()`. But it calls `SubListWithProposer()` inside by using `valSet.GetProposer()`. So basically proposer address is required to get a committee. 

I think if we don't change the logic of making a committee, `SubListWithoutPropser` is not needed.